extremeflash/ws_ap3825i.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+#
+#     Copyright (C) 2023 Grische
+#
+#     This program is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, version 3.
+#
+#     This program is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-only
+"""This tool allows flashing Enterasys WS-AP3825i access points fully automatically, using OpenWRT's initramfs image."""
+import ipaddress
+import logging
+import pathlib
+import re
+import time
+from threading import Thread
+from typing import Optional
+
+import serial
+
+from .tftp_server import TftpServer
+from . import ws_ap3710i
+from .ws_ap3710i import (
+    boot_set_ips,
+    boot_wait_for_brlan,
+    bootup_interrupt,
+    bootup_login,
+    bootup_login_verification,
+    debug_serial,
+    is_kernel_booting,
+    keep_logging_until_reboot,
+    post_cleanup,
+    readline_from_serial,
+    setting_up_ips,
+    start_ssh,
+    write_to_serial,
+)
+
+#  See .ws_ap3710i file for implementation details
+
+
+def bootup_set_boot_openwrt(ser: serial.Serial):
+    if not ws_ap3710i.event_keep_serial_active.is_set():
+        return
+    ser.write(b"printenv\n")
+    time.sleep(1)
+    printenv_return = ser.read(ser.in_waiting).decode("ascii")
+    debug_serial(printenv_return)
+
+    if "MODEL=AP3825i" not in printenv_return:
+        model = re.search(r"MODEL=(.*)\r\n", printenv_return)
+        raise RuntimeWarning(
+            f"Unexpected Model {None if model is None else model.group(1)} found. Aborting to not harm device."
+        )
+
+    boot_openwrt_params = (
+        b"cp.b 0xEC000000 0x2000000 0x2000000; interrupts off; bootm start 0x2000000; bootm loados; "
+        b"fdt resize; fdt boardsetup; fdt chosen; bootm prep; bootm go;"
+    )
+    # TODO: Use better parameters
+    # https://forum.darmstadt.freifunk.net/t/flashing-of-the-extreme-networks-ws-ap3825i/923
+
+    if "boot_openwrt" in printenv_return:
+        logging.debug("Found existing U-Boot boot_openwrt parameter. Verifying.")
+        existing_boot_openwrt_params = re.search(r"boot_openwrt=(.*)\r\n", printenv_return)
+        if not existing_boot_openwrt_params:
+            raise RuntimeError("Unable to parse detected boot_openwrt paramter")
+
+        if boot_openwrt_params.decode("ascii") != existing_boot_openwrt_params.group(1):
+            logging.warning(f"Overwriting unexpected param for 'boot_openwrt': {existing_boot_openwrt_params}")
+        else:
+            logging.debug("Existing U-Boot boot_openwrt parameter looks good.")
+
+    if True:
```

Comment:
I think you want to remove always True condition.

Reply:
There was actually a bug there. I removed the `if True` and added a `return` to the previous `else` above this.
Thanks for spotting!